sampling/include/var_opt_union_impl.hpp

```diff
@@ -508,9 +514,8 @@ void var_opt_union<T, A>::mark_moving_gadget_coercer(var_opt_sketch<T, A>& sk) c
   uint32_t result_r = 0;
   size_t next_r_pos = result_k; // = (result_k+1)-1, to fill R region from back to front
 
-  typedef typename std::allocator_traits<A>::template rebind_alloc<double> AllocDouble;
-  double* wts = AllocDouble().allocate(result_k + 1);
-  T* data     = A().allocate(result_k + 1);
+  double* wts = AllocDouble(allocator_).allocate(result_k + 1);
+  T* data     = A(allocator_).allocate(result_k + 1);
```

Comment:
why not just allocator_.allocate()?

Reply:
There were some places where it was unhappy doing that in a const function

Reply:
right

Reply:
Ok, just checked and the specific compiler error here is that it says there's no allocate() method defined in our test.

Reply:
Sorry, read the whole error more carefully and it didn't match due to const issues.